### PR TITLE
Remove repr from the filters.

### PR DESCRIPTION
**Pull request by jarovo**
Remove repr from the filter construction.
Make filters testable.
Add tests

### CODE REVIEW

**Comment by simon3z** [comment]
@mfalesni @dkorn can you review?

**Comment by mfalesni** [comment]
- sqlfilter is removed and no longer used
- I looked in the source code of manageiq and prepared correct escaping in #9 for filter[]
- I will also add the unicode in there.

**Comment by simon3z** [comment]
Ping @dkorn

File: tests/test_utils.py [hunk 1]
```diff
@@ -1,15 +1,19 @@
...
-    assert '"\'\'\'"' == escape_filter("'''")
-    assert "'\"'" == escape_filter('"')
-    assert "'\"\"'" == escape_filter('""')
-    assert "'\"\"\"'" == escape_filter('"""')
-    assert '1' == escape_filter(1)
-    assert 'NULL' == escape_filter(None)
-    assert "''" == escape_filter('')
+class TestUtils(TestCase):
+    def test_escape_filter(self):
+        self.assertEqual('"\'"', escape_filter("'"))
```
Comment (mfalesni) [inline tests/test_utils.py:9]: No, do not use this in pytest.

File: src/miqclient/api.py [hunk 2]
```diff
@@ -13,6 +13,8 @@
 from functools import partial
 from wait_for import wait_for
 
+from .filters import gen_filter, Q
```
Comment (dkorn) [inline src/miqclient/api.py:16]: unused import `gen_filter`

File: src/miqclient/filters.py [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+        api.collections.vms.filter(Q('name', '=', 'foo') | Q('name', '=', 'bar'))
+
+    Et cetera ... You can use ``|`` and ``&`` operators and they only work in chain (no parentheses)
+    because that is what filter supports.
+
+    """
+    @classmethod
+    def from_dict(cls, d):
+        """Creates a query (AND and =) from a dictionary."""
```
Comment (dkorn) [inline src/miqclient/filters.py:38]: Are you assuming AND between every key, value pair in the dictionary? If so, why not allowing OR?

File: src/miqclient/utils.py [hunk 4]
```diff
@@ -0,0 +1,81 @@
+# -*- coding: utf-8 -*-
+import six
+
+QUOTES = {u'"', u"'"}
+
+
+def give_another_quote(q):
+    """When you pass a quote character, returns you an another one if possible"""
```
Comment (dkorn) [inline src/miqclient/utils.py:8]: nit: drop the 'an'

File: src/miqclient/filters.py [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+        self.op = op
+        self.value = value
+        self.preceeding = None
+        self.is_or = False
+
+    def _set_preceeding(self, preceeding):
+        self.preceeding = preceeding
+
+    @property
+    def as_filters(self):
```
Comment (dkorn) [inline src/miqclient/filters.py:59]: consider adding tests for `as_filters` method

File: src/miqclient/filters.py (hunk 3 above)
Comment (mfalesni) [reply src/miqclient/filters.py:38]: This is used for processing kwargs input which looks like (a=1, b=2) therefore the only semantic sense is `and` IMO

File: src/miqclient/filters.py (hunk 5 above)
Comment (mfalesni) [reply src/miqclient/filters.py:59]: Thse are commits probably cherrypicked from my PR #9 I have the tests there ...

Comment (mfalesni) [reply src/miqclient/filters.py:59]: Here: https://github.com/ManageIQ/manageiq-api-client-python/pull/9/files#diff-f6d69c2d8f0a700a18b486f4b905599dR1

**Pull request closed without merging**